src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ProcessInitialDataSetVersionFunction.cs

```diff
@@ -58,6 +85,67 @@ public class ProcessInitialDataSetVersionFunction(PublicDataDbContext publicData
         }
     }
 
+    [Function(nameof(ImportMetadata))]
+    public async Task ImportMetadata(
+        [ActivityTrigger] Guid dataSetVersionId,
+        Guid instanceId,
+        CancellationToken cancellationToken)
+    {
+        var dataSetVersion = await GetDataSetVersion(
+            dataSetVersionId: dataSetVersionId,
+            instanceId: instanceId,
+            cancellationToken);
+
+        var dataSetVersionImport = dataSetVersion.Imports.Single(i => i.InstanceId == instanceId);
+
+        dataSetVersionImport.Stage = DataSetVersionImportStage.ImportingMetadata;
```

Comment:
I think you are right, this makes things a little neater.

There is one thing which is pretty weird now, this leaves us without needing any Activity trigger input parameter binding and it looks like this scenario hasn't really been catered for Durable functions in the isolated worker.

Prior to using the isolated worker model we would have been able to include a context parameter using `[ActivityTrigger] IDurableActivityContext activityContext`.

For now I have made the `instanceId` both an input parameter and the activity binding:

```csharp
    [Function(nameof(WriteDataFiles))]
    public async Task WriteDataFiles(
        [ActivityTrigger] Guid instanceId,
        CancellationToken cancellationToken)
```

```
await context.CallActivityAsync(name, input: context.InstanceId);
```

In theory we don't need to be passing this around as the input, because as we saw before when `Guid dataSetVersionId` was our only input, Activity functions which have an extra parameter `string instanceId` would have that injected automatically. It's undocumented but this was also working for us with `Guid instanceId` too.

Reply:
Edit: in the latest changes I'm avoiding unnecessarily passing the `context.instanceId` as the input to each function since it's already supplied as mentioned above but this means there's an unused `input` param on every function like so:

```
[ActivityTrigger] object? input
```

Reply:
Resolved in more recent discussion.